### PR TITLE
Twitter to X icon and labels

### PR DESCRIPTION
**Pull request by javier-menendez**
## Purpose

Change the Twitter icon to the X icon and update locales labels from Twitter to X

## See

image
image

### CODE REVIEW

**Review by simi — APPROVED** [review]
Looks good!

**Comment by javier-menendez** [comment]
@simi the footer for emails change was missing, can you check the changes again :pray: 

image

File: test/integration/password_reset_test.rb [hunk 1]
```diff
@@ -207,7 +207,7 @@ def forgot_password_with(email)
 
     visit edit_profile_path
 
-    fill_in "Username", with: "username"
+    fill_in "user_handle", with: "username"
```
Comment (simi) [inline test/integration/password_reset_test.rb:210]: :thinking: What's the reason for this change and change in profile_test?

Comment (javier-menendez) [reply test/integration/password_reset_test.rb:210]: Before this PR the Twitter label in the profile form didn't have a locale: [twitter-label](https://github.com/rubygems/rubygems.org/pull/4570/files#diff-6091542e4570d68ca546a4a893b0d0202a75b9aa6fabf4e8aa0b4122ae6a3544R29), since now the label is Username for locale `en.yml` because it doesn't make sense to put in the form the X logo and then X Username. As a result, now there are going to be two `Username` labels, so the `fill_in` raises with an `Ambiguous match` and that's why I changed to use the `id`

image

**Review by simi — APPROVED** [review]
Looks good @javier-menendez. Calling other 👀 for final ✅ @martinemde @segiddins.
